PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1

```diff
@@ -2208,6 +2209,53 @@ function Get-SddcDiagnosticInfo
             Get-StorageEnclosure -CimSession $AccessNode -StorageSubSystem $Subsystem |
                 Export-Clixml ($Path + "GetStorageEnclosure.XML") }
         catch { Show-Error("Unable to get Enclosures. `nError="+$_.Exception.Message) }
+        
+        # SDDC information
+
+        Show-Update "SDDC resource information"
+
+        try
+        {
+            Get-CimInstance -Namespace "root\SDDC\Management" -ClassName SDDC_Drive  | Export-Clixml ($Path + "GetSddcDrive.XML");
+        }
+        catch { Show-Warning("Unable to get SDDC drive. `nError="+$_.Exception.Message) }
+
+        try
+        {
+            Get-CimInstance -Namespace "root\SDDC\Management" -ClassName SDDC_Server | Export-Clixml ($Path + "GetSddcServer.XML");
+        }
+        catch { Show-Warning("Unable to get SDDC server. `nError="+$_.Exception.Message) }
+
+        try
+        {
+            Get-CimInstance -Namespace "root\SDDC\Management" -ClassName SDDC_Volume | Export-Clixml ($Path + "GetSddcVolume.XML");
+        }
+        catch { Show-Warning("Unable to get SDDC volume. `nError="+$_.Exception.Message) }
+
+        try
+        {
+            Get-CimInstance -Namespace "root\SDDC\Management" -ClassName SDDC_Cluster | Export-Clixml ($Path + "GetSddcCluster.XML");
+        }
+        catch { Show-Warning("Unable to get SDDC cluster. `nError="+$_.Exception.Message) }
+
+        try
+        {
+            Get-CimInstance -Namespace "root\SDDC\Management" -ClassName SDDC_VirtualMachine | Export-Clixml ($Path + "GetSddcVM.XML");
+        }
+        catch { Show-Warning("Unable to get SDDC VM. `nError="+$_.Exception.Message) }
+
+        try
+        {
+            Get-CimInstance -Namespace "root\SDDC\Management" -ClassName SDDC_VirtualSwitch | Export-Clixml ($Path + "GetSddcVSwitch.XML");
+        }
+        catch { Show-Warning("Unable to get SDDC Virtual Switch. `nError="+$_.Exception.Message) }

```

Comment:
I'd be considering doing a foreach $classname to avoid cut/paste repetitiveness (construct filename based on it), but this all looks fine.